pyproject.toml

```diff
@@ -87,7 +87,7 @@ dependencies = [
 [project.optional-dependencies]
 inference = [
   "transformers[torch,sentencepiece]==4.39.3",
-  "sentence-transformers>=2.3.1",  # See haystack/nodes/retriever/_embedding_encoder.py, _SentenceTransformersEmbeddingEncoder
+  "sentence-transformers<=3.0.0,>=2.3.1",  # See haystack/nodes/retriever/_embedding_encoder.py, _SentenceTransformersEmbeddingEncoder
```

Comment:
Does `sentence-transformers==3.0.0` break anything?

Reply:
yeah it does, I got a few mypy [issues](https://github.com/deepset-ai/haystack/actions/runs/10469789311/job/28993557056) related to sentence-transformers changing init parameters for CrossEncoder and SentenceTransformers. Perhaps it can be dealt with differently but this seemed as a reasonable choice as sentence-transformers will likely even further diverge in the future.

Reply:
I understand...
Let's mention this pin and the `python-pptx` pin in the release notes, to inform users.